src/components/views/voip/CallView/CallViewHeader.tsx

```diff
@@ -0,0 +1,135 @@
+    action: 'view_room',
+    room_id: roomId,
+    });
+};
+
+type CallControlsProps = Pick<CallViewHeaderProps, 'pipMode' | 'type'> & {
+  roomId: string;
+};
+function CallControls({ pipMode = false, type, roomId }: CallControlsProps) {
+    return <div className="mx_CallView_header_controls">
+        { (pipMode && type === CallType.Video) &&
+      <div className="mx_CallView_header_button mx_CallView_header_button_fullscreen"
+          onClick={onFullscreenClick}
+          title={_t("Fill Screen")}
+      /> }
+        { pipMode && <div className="mx_CallView_header_button mx_CallView_header_button_expand"
+            onClick={() => onExpandClick(roomId)}
+            title={_t("Return to call")}
+        /> }
+    </div>;
+}
+function SecondaryCallInfo({ callRoom }: {callRoom: Room}) {
+    return <span className="mx_CallView_header_secondaryCallInfo">
+        <AccessibleButton element='span' onClick={() => onRoomAvatarClick(callRoom.roomId)}>
+            <RoomAvatar room={callRoom} height={16} width={16} />
+            <span className="mx_CallView_secondaryCall_roomName">
+                { _t("%(name)s on hold", { name: callRoom.name }) }
+            </span>
+        </AccessibleButton>
+    </span>;
+}
+
+function getAvatarBasedOnRoomType(roomOrWidget: Room | IApp) {
+    if (roomOrWidget instanceof Room) {
+        return <RoomAvatar room={roomOrWidget} height={32} width={32} />;
+    } else if (!isUndefined(roomOrWidget)) {
+        return <WidgetAvatar app={roomOrWidget} height={32} width={32} />;
+    }
+    return null;
+}
+
+export function CallViewHeader({
+  type,
+  pipMode = false,
+  callRooms = [],
+  app,
+  onPipMouseDown,
+}: CallViewHeaderProps) {
+    const [callRoom, onHoldCallRoom] = callRooms;
+    const callTypeText = callTypeTranslationByType[type](app);
+    const avatar = getAvatarBasedOnRoomType(callRoom ?? app);
+    const callRoomName = type === 'widget' ? callTypeText : callRoom.name;
+    const roomId = app ? app.roomId : callRoom.roomId;
+    if (!pipMode) {
+        return <div className="mx_CallView_header">
+            <div className="mx_CallView_header_phoneIcon" />
+            <span className="mx_CallView_header_callType">{ callTypeText }</span>
+            <CallControls roomId={roomId} pipMode={pipMode} type={type} />
+        </div>;
+    }
+    return (<div
+        className="mx_CallView_header"
+        onMouseDown={onPipMouseDown}
+    >
+        <AccessibleButton onClick={() => onRoomAvatarClick(roomId)}>
+            { avatar }
+        </AccessibleButton>
+        <div className="mx_CallView_header_callInfo">
+            <div className="mx_CallView_header_roomName">{ callRoomName }</div>
+            <div className="mx_CallView_header_callTypeSmall">
+                { callTypeText }
+                { onHoldCallRoom && <SecondaryCallInfo callRoom={onHoldCallRoom} /> }
+            </div>
+        </div>
+        <CallControls roomId={roomId} pipMode={pipMode} type={type} />
```

Comment:
I think it would be clearer if this was names a bit differently. Reading this for the first time I though it were the call controls from the Call View

Reply:
Those are call controls from the `CallView`, but I'm not 100% sure if I have moved everything out of there. Seems to be working correctly, but I think you have added some new things there like last week?

Reply:
> Seems to be working correctly, but I think you have added some new things there like last week?

Everything should be there, just icon needs fixing, see above